src/main/java/uk/gov/laa/ccms/caab/controller/application/summary/EditProceedingsAndCostsSectionController.java

```diff
@@ -1271,4 +1286,292 @@ public String caseCostsPost(
+   * @return the view name for prior authority type selection.
+   */
+  @GetMapping("/application/prior-authorities/add/type")
+  public String priorAuthorityType(
+      final Model model) {
+
+    final PriorAuthorityFlowFormData priorAuthorityFlow
+        = new PriorAuthorityFlowFormData(ACTION_ADD);
+
+    model.addAttribute(PRIOR_AUTHORITY_FLOW_FORM_DATA, priorAuthorityFlow);
+    model.addAttribute("priorAuthorityTypeDetails",
+        priorAuthorityFlow.getPriorAuthorityTypeFormDataDetails());
+
+    populatePriorAuthorityTypeDropdown(model);
+
+    return "application/prior-authority-type";
+  }
+
+  private void populatePriorAuthorityTypeDropdown(final Model model) {
+    final List<PriorAuthorityTypeDetail> priorAuthorityTypes = Optional.ofNullable(
+        lookupService.getPriorAuthorityTypes().block())
+        .map(PriorAuthorityTypeDetails::getContent)
+        .orElse(Collections.emptyList());
+
+    model.addAttribute("priorAuthorityTypes", priorAuthorityTypes);
+  }
+
+  /**
+   * Handles submission of the prior authority type form, updating the session and redirecting
+   * to details.
+   *
+   * @param priorAuthorityFlow the prior authority flow data.
+   * @param priorAuthorityTypeDetails the selected prior authority type details.
+   * @param model the model to add attributes to for the view.
+   * @param bindingResult the result of form validation.
+   * @return the redirect URL for the prior authority details page.
+   */
+  @PostMapping("/application/prior-authorities/add/type")
+  public String priorAuthorityTypePost(
+      @SessionAttribute(PRIOR_AUTHORITY_FLOW_FORM_DATA)
+      final PriorAuthorityFlowFormData priorAuthorityFlow,
+      @ModelAttribute("priorAuthorityTypeDetails")
+      final PriorAuthorityFormDataTypeDetails priorAuthorityTypeDetails,
+      final Model model,
+      final BindingResult bindingResult) {
+
+    priorAuthorityTypeValidator.validate(priorAuthorityTypeDetails, bindingResult);
+
+    if (bindingResult.hasErrors()) {
+      populatePriorAuthorityTypeDropdown(model);
+      model.addAttribute(PRIOR_AUTHORITY_FLOW_FORM_DATA, priorAuthorityFlow);
+      model.addAttribute("priorAuthorityTypeDetails", priorAuthorityTypeDetails);
+      return "application/prior-authority-type";
+    }
+
+    priorAuthorityFlow.setPriorAuthorityTypeFormDataDetails(priorAuthorityTypeDetails);
+    model.addAttribute(PRIOR_AUTHORITY_FLOW_FORM_DATA, priorAuthorityFlow);
+
+    return "redirect:/application/prior-authorities/add/details";
+  }
+
+  /**
+   * Displays the prior authority details page for adding or editing.
+   *
+   * @param priorAuthorityAction the action (add or edit) being performed.
+   * @param priorAuthorityFlow the flow data for prior authority actions.
+   * @param model the model for adding attributes to the view.
+   * @return the view name for prior authority details.
+   */
+  @GetMapping("/application/prior-authorities/{action}/details")
```

Comment:
It looks a bit wrong having the action type in the middle of the path. Should it not be at the end?

Reply:
This is the same as how proceedings and cost is done